lib/Reader/reader.cpp

```diff
@@ -7158,6 +7161,12 @@ void ReaderBase::readBytesForFlowGraphNodeHelper(
       getCallInfo(&ResolvedToken, nullptr /*constraint*/,
                   CORINFO_CALLINFO_LDFTN, &CallInfo);
 
+      // ReadyToRun has delegate constructor optimization for non-virtual
+      // function pointers resolved at compile time only for now.
+      if ((Flags & CORJIT_FLG_READYTORUN) && (CallInfo.kind != CORINFO_CALL)) {
```

Comment:
Not sure I understand the impact of "for now" here.... do you mean this optimization is going away, or that in RtR it is currently limited in scope but might become more widely capable?